### PR TITLE
converting to explicit ruby version built from source

### PR DESCRIPTION
**Pull request by JoeNunnelley**

### CODE REVIEW

File: ruby/2.3/Dockerfile [hunk 1]
```diff
@@ -1,17 +1,114 @@
...
+ENV BUNDLER_VERSION 1.16.1
+
+# some of ruby's build scripts are written in ruby
+#   we purge system ruby later to make sure our final image uses what we just built
+RUN set -ex \
+	\
+	&& buildDeps=' \
+		bison \
+		ruby \
+    wget \
```
Comment (michaelb990) [inline ruby/2.3/Dockerfile]: fix indentation.

File: ruby/2.3/Dockerfile [hunk 2]
```diff
@@ -1,17 +1,114 @@
...
+		libssl-dev \
+		libtool \
+		libwebp-dev \
+		libxml2-dev \
+		libxslt-dev \
+		libyaml-dev \
+		make \
+		patch \
+		xz-utils \
+    zlib1g-dev \
```
Comment (michaelb990) [inline ruby/2.3/Dockerfile]: fix indentation.

File: runit-ruby/2.3/Dockerfile [hunk 3]
```diff
@@ -1,17 +1,114 @@
...
+ENV BUNDLER_VERSION 1.16.1
+
+# some of ruby's build scripts are written in ruby
+#   we purge system ruby later to make sure our final image uses what we just built
+RUN set -ex \
+	\
+	&& buildDeps=' \
+		bison \
+		ruby \
+    wget \
```
Comment (michaelb990) [inline runit-ruby/2.3/Dockerfile]: fix indentation.

**Comment by cyranix** [comment]
🎉  This is definitely a good step to take.

Is there a way to make this less backslashy, by factoring a big chunk of the change into a shell script that gets executed within the Dockerfile?

**Comment by JoeNunnelley** [comment]
successful build using the runit-ruby-2.3:2.3.5 tagged images

https://jenkins-build.socrata.com/job/AA%20Exploration%20Team/job/Storyteller/job/storyteller/3519/

**Comment by smooshy** [comment]
ditto @cyranix about making it more readable... but a great step to take.

**Pull request merged**